qualtricssurvey/models.py

```diff
@@ -663,29 +657,50 @@ def end_survey(self, data, suffix=''):  # pylint: disable=unused-argument
         }
         return response
 
+    def publish_grade(self):
+        grade_dict = {
+            'value': self.score.raw_earned,
+            'max_value': self.score.raw_possible,
+        }
+        LOGGER.info("Sanders", self.score)
+        self.runtime.publish(self, "grade", grade_dict)
+
     def has_submitted_answer(self):
         """
         Currently unused.
         """
-        return self.done
+        try:
+            survey_status = SurveyStatus.objects.get(usage_key=self.location, user_id=self.xmodule_runtime.user_id).status         
```

Comment:
@bsande6 I would set the field `is_answered` in the `end_survey` handler or whenever the instructor does a reset of their attempts then just perform a `return self.is_answered` here and remove the check from the database for `SurveyStatus.status`.